projects/storefrontapp-e2e-cypress/cypress/helpers/quote.ts

```diff
@@ -84,17 +108,205 @@ export function submitQuote(): void {
+export function validateItemQuantity(
+  itemIndex: number,
+  expectedQuantity: string
+): void {
+  log(validateItemQuantity.name);
+  cy.get(
+    `cx-quote-details-cart .cx-item-list-row:nth-child(${itemIndex})`
+  ).within(() => {
+    cy.get('cx-item-counter input').should('have.value', expectedQuantity);
+  });
+}
+
+/**
+ * Click the 'Remove' button for the item at the given index in the QDP cart
+ * @param itemIndex Index of the Item in the QDP cart list
+ */
+export function removeItemOnClick(itemIndex: number): void {
+  log(removeItemOnClick.name);
+  cy.get(
+    `cx-quote-details-cart .cx-item-list-row:nth-child(${itemIndex})`
+  ).within(() => {
+    cy.get('button').contains('Remove').click();
+  });
+  cy.get<string>('@quoteURL').then(cy.visit);
+}
+
+/**
+ * Checks if the given Item exists within the QDP cart at the given index.
+ * @param itemIndex Index of the Item in the QDP cart list
+ * @param productID Id of the product
+ * @param itemExpected 'true' if the item should be there, 'false' if no item should exist.
+ */
+export function checkItemAtIndexExists(
+  itemIndex: number,
+  productID: string,
+  itemExpected: boolean
+): void {
+  log(checkItemAtIndexExists.name);
+  if (itemExpected) {
+    cy.get(`cx-quote-details-cart .cx-item-list-row:nth-child(${itemIndex})`)
+      .should('be.visible')
+      .contains(productID);
+  }
+  if (itemIndex === 1 && !itemExpected) {
+    cy.get(
+      `cx-quote-details-cart .cx-item-list-row:nth-child(${itemIndex})`
+    ).should('not.exist');
+  }
+}
+
+/**
+ * Checks if the "Quote Information" card is in edit mode
+ * @param cardTitle Title of the card
+ * @param cardEditModeActive Indicates if the card is in edit mode
+ */
+export function verifyQuoteInformationCardInEditMode(
+  cardTitle: string,
+  cardEditModeActive: boolean
+): void {
+  log(verifyQuoteInformationCardInEditMode.name);
+  cy.get(`cx-quote-details-overview .cx-container .card-body`)
+    .contains(cardTitle)
+    .should('exist')
+    .then(() => {
+      if (cardEditModeActive) {
+        cy.get('button').contains('Save').should('exist');
+      } else {
+        cy.get('button').contains('Safe').should('not.exist');
+      }
+    });
+}
+
+/**
+ * Changes the "Quote Name" and the "Quote description" to given values
+ * @param cardTitle Title of the card
+ * @param newQuoteName New quote name
+ * @param newQuoteDescription New quote description
+ */
+export function changeQuoteSummaryCardEntries(
+  cardTitle: string,
+  newQuoteName: string,
+  newQuoteDescription: string
+): void {
+  log(changeQuoteSummaryCardEntries.name);
+  cy.get(`cx-quote-details-overview .cx-container .card-body`)
+    .contains(cardTitle)
+    .should('exist')
+    .then(() => {
+      cy.get(`cx-quote-details-overview .cx-container .card-body input`)
+        .should('exist')
+        .clear()
+        .type(newQuoteName);
+      cy.get(`cx-quote-details-overview .cx-container .card-body textarea`)
+        .should('exist')
```

Comment:
do we need these 3 .should('exist') here?
we could just assume they exists,if not the test fails anyways.

Reply:
agreed